.github/workflows/docs-ci.yml

```diff
@@ -36,8 +36,8 @@ jobs:
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
           report_changed_scope_only: false
-          fail_on_warnings: true
-          fail_on_error: true
```

Comment:
These were intentionally turn on to keep the documentation in good shape.
We've fixed all past issues for a while now, let's keep it that way so we don't fall behind again and also have annoying warnings/errors polluting every single PR.

Reply:
the check is not working correctly for ESM imports. it also prevents us from moving fast. if it's polluting all of our PRs, lets shut the check off.

Reply:
what value is there in this being done because you're trying to satisfy a CI check so you can push code?

```
/** gets the permission grants */
getPermissionGrants()
```

i have seen this all over the repo and in some cases comments written by you because you're satisfying the check. i don't want to do this, it's double the code.

Reply:
I think it's fair to bring up the additional burden and the fact that some comments are definitely unnecessary and only satisfying the bot (I'm guilty of this!)

But it seems out of scope for this PR, we should have that discussion and decide on better CI rules on their own merit.

Reply:
When we have that discussion i would love for one of us to go through and delete useless comments because they seriously take away from readability which is why I didn't want to ship them

Reply:
I feel the same pain and agree on some comments being pretty useless, and did not have time and passionate enough to sink on challenging the intent of the original owner/creator that sets up the doc rule and CICD.

The error/warning pollution in PRs was however distracting and annoying enough that I followed the existing intent and cleanup all existing warnings/errors, wasn't too painful as frankly Chad did most of the work.

I do agree with @LiranCohen that this is out of scope though happy to discuss about this in a separate issue PR. If you are passionate about this, feel free to create an issue and assign it to yourself to drive to resolution.

But before we reach consensus on changing the rules, we should follow them.

Reply:
Made a ticket for it. Keep in mind it's not working for this PR so its checks will need to be bypassed with an admin merge (or set back to warning as I did in the PR)

Reply:
Pretty sure it is working as intended. Happy to push a change to fix these after you are done with all the force-pushes so my fixes don't get overwritten.

Reply:
once this is approved and then #830 is approved and merged into this I can revert this change

Reply:
Word. Please do revert when #830 goes in. 🙏